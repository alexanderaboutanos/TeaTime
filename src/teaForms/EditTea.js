/** @format */

import React, { useState, useEffect } from "react";
import { useNavigate, useParams } from "react-router-dom";
import TeaTimeApi from "../api/teaTime.api";
import Alert from "../common/Alert";

/** EditTea
 *
 * Shows form and manages update to state on changes.
 * On submission:
 * - calls signup function prop
 * - redirects to /companies route
 *
 * Routes -> EditTea -> Alert
 * Routed as /signup
 */

function EditTea() {
  const navigate = useNavigate();
  const { id } = useParams();
  const [formData, setFormData] = useState({
    title: "",
    brand: "",
    description: "",
    category: "",
    review: "",
    country_of_origin: "",
    organic: false,
    img_url: "",
    brew_time: null,
    brew_temp: null,
  });
  const [formErrors, setFormErrors] = useState([]);

  useEffect(() => {
    async function getTeaDataOnMount() {
      const tea = await TeaTimeApi.getTeaInfo(id);
      console.log(tea);
      setFormData(tea);
    }
    getTeaDataOnMount();
  }, []);

  console.debug("EditTea, ", "formData=", formData, "formErrors=", formErrors);

  /** Handle form submit:
   *
   * Calls func prop and, if successful, redirect to /my-teas.
   */

  async function handleSubmit(evt) {
    evt.preventDefault();
    try {
      delete formData.id;
      if (formData.img_url === "") formData.img_url = null;
      await TeaTimeApi.editTea(id, formData);
      alert("edited your tea!");
      navigate(`/tea/${id}`, { replace: true });
    } catch (errors) {
      setFormErrors(errors);
    }
  }

  /** Update form data field */
  function handleChange(evt) {
    console.log("EVT: ", evt);
    const target = evt.target;
    let value;
    if (target.type === "checkbox") value = target.checked;
    else if (target.type === "number") value = parseInt(target.value);
    else value = target.value;
    const name = target.name;
    setFormData((data) => ({ ...data, [name]: value }));
  }

  return (
    <div className="EditTea">
      <div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
        <h2 className="mb-3">Edit Your Tea</h2>
        <div className="card">
          <div className="card-body">
            <form onSubmit={handleSubmit}>
              <div className="form-group">
                <label>Title</label>
                <input
                  name="title"
                  className="form-control"
                  value={formData.title}
                  onChange={handleChange}
                />
              </div>
              <div className="form-group">
                <label>Brand</label>
                <input
                  name="brand"
                  className="form-control"
                  value={formData.brand}
                  onChange={handleChange}
                />
              </div>
              <div className="form-group">
                <label>Description</label>
                <input
                  name="description"
                  className="form-control"
                  value={formData.description}
                  onChange={handleChange}
                />
              </div>
              <div className="form-group">
                <label>Category</label>
                <input
                  name="category"
                  className="form-control"
                  value={formData.category}
                  onChange={handleChange}
                />
              </div>
              <div className="form-group">
                <label>Review</label>
                <input
                  name="review"
                  className="form-control"
                  value={formData.review}
                  onChange={handleChange}
                />
              </div>
              <div className="form-group">
                <label>Country of Origin</label>
                <input
                  name="country_of_origin"
                  className="form-control"
                  value={formData.country_of_origin}
                  onChange={handleChange}
                />
              </div>
              <div className="form-group">
                <label>Link to image of tea</label>
                <input
                  type="text"
                  name="img_url"
                  className="form-control"
                  value={formData.img_url}
                  onChange={handleChange}
                />
              </div>
              <div className="form-group">
                <label>Brew Time (in minutes)</label>
                <input
                  type="number"
                  name="brew_time"
                  className="form-control"
                  value={formData.brew_time}
                  onChange={handleChange}
                />
              </div>
              <div className="form-group">
                <label>Brew Temp (Â°F)</label>
                <input
                  type="number"
                  name="brew_temp"
                  className="form-control"
                  value={formData.brew_temp}
                  onChange={handleChange}
                />
              </div>
              <div className="form-group">
                <label>Organic Tea?</label>
                <input
                  type="checkbox"
                  name="organic"
                  value={formData.organic}
                  checked={formData.organic}
                  onChange={handleChange}
                />
              </div>

              {formErrors.length ? (
                <Alert type="danger" messages={formErrors} />
              ) : null}

              <button
                type="submit"
                className="btn btn-primary float-right"
                onSubmit={handleSubmit}
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
}

export default EditTea;
