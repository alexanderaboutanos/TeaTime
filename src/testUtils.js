/** @format */

import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser1",
  first_name: "Test",
  last_name: "User",
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

const fakeTeas = [
  {
    id: 740259,
    title:
      "Tea Sampler - Fruit Tea - Antioxidants Tea - Rooibos Tea - 100% Natural - Decaffeinated - Blueberry Tea - Peach Tea - Raspberry Tea - Loose Leaf Tea",
    brand: "Chinese Tea Culture",
    description:
      "Tea Sampler - Fruit Tea - Antioxidants Tea - Rooibos Tea - 100% Natural - Decaffeinated - Blueberry Tea - Peach Tea - Raspberry Tea - Loose Leaf Tea",
    img_url: "https://spoonacular.com/productImages/740259-312x231.jpeg",
  },
  {
    id: 1539729,
    title:
      "Tea Forte Lotus Relaxing Teas Presentation Box Tea Sampler Gift Set, 20 Assorted Variety Handcrafted Pyramid Tea Infuser Bags, Black Tea, Green Tea, Oolong Tea, White Tea, Herbal Tea",
    brand: "Tea Forte",
    description:
      "LOTUS COLLECTION, a curated assortment of five relaxing hot teas for mind, body & spirit - a unique and unprecedented gourmet collection of highest-grade, small-lot, certified organic teas: Vanilla Pear, Orange Jasmine, Mountain Oolong, Darjeeling Quince, Lemon Lavender DIVERSE, SUBTLE FLAVORS from only the finest whole tea leaves in the world, blended with aromatic fruits, flowers and herbs DELIGHTFUL TEA GIFT SET offers a variety of premium gourmet teas, a most welcomed hostess gift or gift for tea lovers PRESENTATION BOX tea sampler has 20 assorted pyramid tea bag infusers, handcrafted to allow the delicate leaves to luxuriantly unfurl in hot water, producing a deliciously aromatic, flavorful cup WRAPPED IN ELEGANT, embossed pattern gift box with a satin ribbon, open the lid to reveal a tea tasting menu with blend descriptions",
    img_url: "https://spoonacular.com/productImages/1539729-312x231.jpeg",
  },
];

export { UserProvider, fakeTeas };
