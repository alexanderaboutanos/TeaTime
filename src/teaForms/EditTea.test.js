/** @format */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import EditTea from "./EditTea";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <EditTea />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <EditTea />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
