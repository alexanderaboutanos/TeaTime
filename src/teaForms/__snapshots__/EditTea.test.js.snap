// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Edit"
  >
    <div
      class="container"
    >
      <h2
        class="mb-3"
        style="margin: 20px;"
      >
        Edit your tea
      </h2>
      <p>
        <i>
          Please adjust the tea information below
        </i>
      </p>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group row"
              style="margin: 5px;"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Title:
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control"
                  name="title"
                  placeholder="required"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group row"
              style="margin: 5px;"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Brand:
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control"
                  name="brand"
                  placeholder="required"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group row"
              style="margin: 5px;"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Description:
              </label>
              <div
                class="col-sm-10"
              >
                <input
                  class="form-control"
                  name="description"
                  placeholder="required"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group row"
              style="margin: 5px;"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Category:
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control"
                  name="category"
                  placeholder="Black, Green, Oolong, etc."
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group row"
              style="margin: 5px;"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Country of Origin:
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control"
                  name="country_of_origin"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group row"
              style="margin: 5px;"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Review:
              </label>
              <div
                class="col-sm-10"
              >
                <input
                  class="form-control"
                  name="review"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group row"
              style="margin: 5px;"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Link to image of tea:
              </label>
              <div
                class="col-sm-10"
              >
                <input
                  class="form-control"
                  name="img_url"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group row"
              style="margin: 5px;"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Brew Time (min):
              </label>
              <div
                class="col-sm-2"
              >
                <input
                  class="form-control"
                  name="brew_time"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group row"
              style="margin: 5px;"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Brew Temp (Â°F):
              </label>
              <div
                class="col-sm-2"
              >
                <input
                  class="form-control"
                  name="brew_temp"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group row"
              style="margin: 5px;"
            >
              <label
                class="col-sm-2 col-form-label"
              >
                Organic Tea?
              </label>
              <div
                class="col-sm-1"
              >
                <input
                  name="organic"
                  type="checkbox"
                  value="false"
                />
              </div>
            </div>
            <button
              class="btn btn-primary btn-lg"
              type="submit"
            >
              Edit Your Tea!
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
